server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -2583,6 +2598,21 @@ private void createDefaultEgressFirewallRule(final List<FirewallRule> rules, fin
         }
     }
 
+    private void createDefaultEgressIpv6FirewallRule(final List<FirewallRule> rules, final long networkId) {
+        final NetworkVO network = _networkDao.findById(networkId);
+        if(!_networkOfferingDao.isIpv6Supported(network.getNetworkOfferingId())) {
+            return;
+        }
+        // Since not all networks will IPv6 supported, add a system rule for IPv6 networks
+        final List<String> sourceCidr = new ArrayList<String>();
+        final List<String> destCidr = new ArrayList<String>();
+        sourceCidr.add(network.getIp6Cidr());
+        destCidr.add(NetUtils.ALL_IP6_CIDRS);
+        final FirewallRule rule = new FirewallRuleVO(null, null, null, null, "all", networkId, network.getAccountId(), network.getDomainId(), Purpose.Ipv6Firewall, sourceCidr,
```

Comment:
All egress rules are not allowed by default. It would depend on network's egress policy.
A rule for `all` protocols is added here to make python script in router understand that it needs to configure IPv6 firewall for the network, allow or deny logic based on policy is present in `systemvm/debian/opt/cloud/bin/configure.py`

Reply:
@shwstppr 
OK, that sounds good!
 
ps, 'all' can be replaced with `NetUtils.ALL_PROTO`.